### ISSUE TITLE
[Question] How I customize StreamingLambda with @searchable

### ISSUE THREAD

**Issue by bokuweb** [aws-amplify/amplify-cli#1022]
>**Note**: If your question is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

** Which Category is your question related to? **

AppSync with @searchable

** What AWS Services are you utilizing? **

appsync elasticsearch service

** Provide additional details e.g. code snippets **

Hi :) I would like to customize `ElasticSearchStreamingLambda`?When I add following schema, lambda was created with python code.

```
type Test @model @searchable {
  id: ID!
}
```

So I would like to use nodejs or golang and customize lambda code.
Is it possible now?? I can not find better way  yet...

**Comment by bokuweb** [comment]
Should I create another function by myself then overwrite `ElasticsearchStreamingFunctionName ` (https://aws-amplify.github.io/docs/cli/graphql#elasticsearchstreamingfunctionname) ?

**Comment by kaustavghosh06** [comment]
@bokuweb Yes, you could probably do that as mentioned in https://aws-amplify.github.io/docs/cli/graphql#elasticsearchstreamingfunctionname

**Comment by bokuweb** [comment]
@kaustavghosh06 Thanks for your quick response :) I'll try it

**Comment by kaustavghosh06** [comment]
@bokuweb Were you able to overwrite the parameters?

**Comment by bokuweb** [comment]
@kaustavghosh06 Sorry, I did not it yet. I'll try it and write result here :)

**Comment by kstro21** [comment]
@bokuweb setting https://aws-amplify.github.io/docs/cli/graphql#elasticsearchstreamingfunctionname will only change the name of the Lambda, but you can not override the code, it will still use the same zip containing the Python code, here is a piece of the template definition

```
"ElasticSearchStreamingLambdaFunction": {
    "Type": "AWS::Lambda::Function",
    "Properties": {
        "Code": {
            "S3Bucket": {
                "Ref": "S3DeploymentBucket"
            },
            "S3Key": {
                "Fn::Join": [
                    "/",
                    [
                        {
                            "Ref": "S3DeploymentRootKey"
                        },
                        "functions",
                        {
                            "Fn::Join": [
                                ".",
                                [
                                    "ElasticSearchStreamingLambdaFunction",
                                    "zip"
                                ]
                            ]
                        }
                    ]
                ]
            }
        },
```

Right now, I don't see any doc that references how to use a custom Lambda instead of the auto-generated one.

You can still copy the autogenerated stack `SearchableStack.json` from the build folder of another project, copy to the _amplify\backend\api\pgaplatform\stacks\_ inside yours, the customize the Lambda. It will require to edit a few files in the process, but it will be ideal that *amplify* support it out of the box.

**Comment by bokuweb** [comment]
@kstro21 Thanks for your reply :)

> will only change the name of the Lambda, but you can not override the code, it will still use the same zip containing the Python code,

Hmmm.It is very disappointing... 😢 

> You can still copy the autogenerated stack SearchableStack.json from the build folder of another project, copy to the _amplify\backend\api\pgaplatform\stacks_ inside yours, the customize the Lambda. It will require to edit a few files in the process, but it will be ideal that amplify support it out of the box.

Thanks 👍 I'll try and do some investigation.

@kaustavghosh06 @mikeparisstuff Do you have any idea??

**Comment by mikeparisstuff** [comment]
@bokuweb This is currently not possible although I agree that this is useful and it would not be too difficult to implement. Currently the ElasticSearchStreamingLambdaFunctionName argument changes the name of the function deployed by Amplify but this is only partially useful. We can add a parameter "ElasticSearchStreamingLambdaFunctionArn" parameter that you could use to point to an existing function. Once adding that we need to make a few tweaks to the EventSourceMapping resource here https://github.com/aws-amplify/amplify-cli/blob/767f3a6edce9f8c4109e18afacf648c6d4c7be56/packages/graphql-elasticsearch-transformer/src/resources.ts#L164-L168. I will try to look into this when I can but would also be happy to review a PR.

**Comment by bokuweb** [comment]
@mikeparisstuff Thanks for your reply :) What should I do specifically? I will try to fix it if possible.

**Comment by kstro21** [comment]
@mikeparisstuff and @kaustavghosh06 wouldn't be easy to allow for stacks in `amplify\backend\api\amplifytest\stacks\` to update stacks with the exact same name in `amplify\backend\api\amplifytest\build\stacks\` by doing a **deep merge**? That will be interesting to see.
